### ISSUE TITLE
no-duplicate-super rule does not validate when using a ternary operator

### ISSUE THREAD

**Issue by Nysosis** [palantir/tslint#3275]
### Bug Report

- __TSLint version__: 5.3.2
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class Base {
	constructor(props?: any) { }
}

class IfElse extends Base {
	constructor(props?: any) {
		if (props) {
			super(props)
		} else {
			super()
		}
	}
}

class Ternary extends Base {
	constructor(props?: any) {
		props ? super(props) : super()
	}
}

```

with `tslint.json` configuration:

```json
{
	"rules": {
		"no-duplicate-super": true
	}
}

```

#### Actual behavior
Class named `Ternary` is failing to pass the `no-duplicate-super` rule

#### Expected behavior
Class named `Ternary` to pass the `no-duplicate-super` rule

**Comment by ajafff** [comment]
Fixed by #3544
